src/map/script.cpp

```diff
@@ -18205,6 +18205,32 @@ BUILDIN_FUNC(delmonsterdrop)
 }
 
 
+
+/*==========================================
+ * Returns a random mob_id
+ * type: Where to fetch from (see enum e_random_monster)
+ * flag: Type of checks to apply (see enum e_random_monster_flags)
+ * lv: Mob level to check against
+ *------------------------------------------*/
+BUILDIN_FUNC(getrandmobid)
+{
+	int type = script_hasdata(st, 2) ? script_getnum(st, 2) : MOBG_BRANCH_OF_DEAD_TREE;
+
+	if (type < MOBG_BRANCH_OF_DEAD_TREE || type >= MOBG_MAX) {
+		ShowWarning("buildin_getrandmobid: Invalid type %d.\n", type);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int flag = script_hasdata(st, 3) ? script_getnum(st, 3) : RMF_MOB_NOT_BOSS;
```

Comment:
Missing check of the flag

Reply:
but its possible to using `RMF_NONE`.
anyway, how do add a check to ensure the flag is valid or declared in enum ?
would this be sufficient ?
`if (flag != NULL)`

Reply:
For now you can make it like you did above
`if (flag < RMF_NONE || flag > RMF_ALL) {` then throw it in mob_get_random_id as enum